### PR TITLE
Don't drop more than three consecutive packets

### PR DESCRIPTION
**Pull request by larseggert**

### CODE REVIEW

**Comment by larseggert** [comment]
I picked three, because if you loose the three initial ones in row, your PTO goes 1->2->4, meaning you waste 7 seconds, and many stacks use very short idle timeouts when they run.

I can look into making the limit configurable.

File: sim/scenarios/drop-rate/drop-rate-error-model.cc [hunk 1]
```diff
@@ -13,16 +13,21 @@ TypeId DropRateErrorModel::GetTypeId(void) {
...
 DropRateErrorModel::DropRateErrorModel()
-    : rate(0), distr(0, 99) {
+    : rate(0), consecutivelyCorrupted(0), distr(0, 99) {
     std::random_device rd;
     rng = new std::mt19937(rd());
 }
 
-void DropRateErrorModel::DoReset(void) { }
- 
+void DropRateErrorModel::DoReset(void) {
```
[the remark this answers is not in the document]
Comment (larseggert) [reply sim/scenarios/drop-rate/drop-rate-error-model.cc]: I don't know - I didn't add it.

**Comment by janaiyengar** [comment]
I can see the value in this, especially since the RNG does this regression to the mean thing where consecutive drops happen occasionally and are annoying. That said, I've found it to be also useful in exposing issues -- stacks running with super low idle timeouts for instance sounds like a stack issue.

I'm concerned that if we _never_ see consecutive drops of > 3 runs, that's going to hide particular classes of bugs.

**Comment by larseggert** [comment]
I'm fine with not merging this, but then we need to make stack images use some sane idle timeout.

**Pull request closed without merging**